### PR TITLE
RegionExtractor mask img

### LINKED ISSUE #1931: regions from RegionExtractor with a given mask_img lie outside mask
Hi all,

I am trying to extract regions that lie within a mask image from dict learning components image but I got regions outside my mask. The following code
```Python
from nilearn.regions import RegionExtractor
extractor = RegionExtractor(components_img, mask_img=mask_img)
extractor.fit()

mask_data = mask_img.get_data()
data = extractor.regions_img_.get_data()
print(data[mask_data == 0, 0].max())
```
outputs 0.0064742849 instead of 0. My nilearn version is 0.5.0.

I can do a PR to fix it

### PR DESCRIPTION
**Pull request by salma1601**
try fix #1931

### CODE REVIEW

**Comment by KamalakerDadi** [comment]
The documentation of mask_img in RegionExtractor should reflect to the changes this PR made.

Also, please add into doc/whats_new.rst about this fix

**Comment by salma1601** [comment]
>Also, please add into doc/whats_new.rst about this fix
Ok. Shall I put it under the 0.5.1 section ?

>The documentation of mask_img in RegionExtractor should reflect to the changes this PR made.
The doc is already OK no ? I did add nothing new, I just fixed the bug that the input mask image was not taken into account

**Comment by kchawla-pi** [comment]
> Ok. Shall I put it under the 0.5.1 section ?

Yeah whichever is the latest, under fixes/bug fix.

**Comment by KamalakerDadi** [comment]
>The doc is already OK no ?

Yes. I think you are right.

File: doc/whats_new.rst [hunk 1]
```diff
@@ -37,6 +37,11 @@ Fixes
 -----
 
 - Example plot_seed_to_voxel_correlation now really saves z-transformed maps.
+- Fix bug in region_extractor.connected_regions which did not systematically
+use the mask_img when provided
+- Fix bug in regions.RegionExtractor which did not use the mask_img when
+provided
+
```
Comment (kchawla-pi) [inline doc/whats_new.rst]: Keeping it consistent with the grammar of other entries, rewrite this as
`region_extractor.connected_regions`  & `regions.RegionExtractor` nowcorrectly use the provided mask_img.

The first letter of each line(not sentence) should be aligned in RestructuredText. In the text in your commit,
the letter 'u' of use should be right under the letter 'F' of Fix.

Comment (kchawla-pi) [reply doc/whats_new.rst]: Let CircleCi complete first, so the next push done the same day will be faster.

**Pull request merged**